src/shapes/textbox.class.js

```diff
@@ -78,6 +86,33 @@
      */
     splitByGrapheme: false,
 
+    /**
+     * While editing handle differently
+     * @private
+     * @param {string} key
+     * @param {*} value
+     */
+    _set: function (key, value) {
+      if (key === 'width') {
+        if (typeof this.maxWidth === 'number' && this.maxWidth < value) {
+          value = this.maxWidth;
+        }
+        if (typeof this.minWidth === 'number' && this.minWidth > value) {
+          value = this.minWidth;
+        }
+      }
+      if (key === 'maxWidth' && this.minWidth > value) {
+        value = this.minWidth;
+      }
+      this.callSuper('_set', key, value);
+      if (key === 'minWidth' && this.maxWidth < value) {
+        this._set('maxWidth', value);
+      }
```

Comment:
this is:
If you want to set maxWidth below minWidth i m going to stop you to minWidth.
If you set up minWidth over maxWidth that is allowed, but  then i will rise maxWidth accordingly.

I see how minWidth should be the boss to avoid creating zero width objects and mess up when the user is driving the action. But this needs to be described too in the properties, let's not have the developers discover it by trial and error.

Reply:
We can warn as well...
What do you prefer?

Reply:
resolved